app/controllers/observations/images_controller.rb

```diff
@@ -0,0 +1,369 @@
+# frozen_string_literal: true
+
+# Clicking on an image currently fires a GET to these actions... because it
+# comes from a link made by thumbnail_helper#thumbnail(link: url_args)
+# with CRUD refactor, change thumbnail helper to fire a POST somehow?
+
+module Observations
+  class ImagesController < ApplicationController
+    before_action :login_required
+    before_action :pass_query_params
+    before_action :disable_link_prefetching, except: [:edit]
+
+    # NEW: Upload Images to an Observation.
+    # was ImageController#add_image
+    def new
+      return unless (@observation = find_observation!)
+
+      return unless check_observation_permission!
+
+      @image = rough_cut_image
+      @licenses = current_license_names_and_ids
+      init_project_vars_for_add_or_edit(@observation)
+    end
+
+    # new (upload) image commits here
+    def create
+      return unless (@observation = find_observation!)
+
+      return unless check_observation_permission!
+
+      create_image
+    end
+
+    private
+
+    def find_observation!
+      find_or_goto_index(Observation, params[:id].to_s)
+    end
+
+    def check_observation_permission!
+      return true if check_permission!(@observation)
+
+      redirect_with_query(permanent_observation_path(id: @observation.id))
+      false
+    end
+
+    def init_project_vars_for_add_or_edit(obs_or_img)
+      @projects = User.current.projects_member(order: :title,
+                                               include: :user_group)
+      @project_checks = {}
+      obs_or_img.projects.each do |proj|
+        @projects << proj unless @projects.include?(proj)
+        @project_checks[proj.id] = true
+      end
+    end
+
+    def current_license_names_and_ids
+      License.current_names_and_ids(@image.license)
+    end
+
+    def rough_cut_image
+      @image = Image.new
+      @image.license = @user.license
+      @image.copyright_holder = @user.legal_name
+      @image.user = @user
+      # Set the default date to the date of the observation
+      # Don't know how to correctly test this.
+      @image.when = @observation.when
+      @image
+    end
+
+    def create_image
+      if params[:upload].blank?
+        flash_warning(:runtime_no_changes.t)
+      else
+        args = params[:image]
+        i = 1
+        while i < 5 || params[:upload]["image#{i}"].present?
+          process_image(args, params[:upload]["image#{i}"])
+          i += 1
+        end
```

Comment:
First time I noticed this. What is this doing? Does it need a comment?

Reply:
I believe: There are five upload spots on the form. This says, for as many inputs as have an image, process the image. 

Adding comment now